test/cmd_line/tabCompletion.test.ts

```diff
@@ -80,4 +78,162 @@ suite('cmd_line tabComplete', () => {
+    const inner10 = await t.createDir(join(inner1, 'inner10'));
+    const inner100 = await t.createDir(join(inner10, 'inner100'));
+
+    // Tab to see the completion of tempDir
+    const cmd = `:e ${tmpDir}${sep}`.split('');
+    await modeHandler.handleMultipleKeyEvents(cmd);
+    await modeHandler.handleKeyEvent('<tab>');
+    let statusBarAfterTab = StatusBar.Get().trim();
+    let expectedPath = `${tmpDir}${sep}inner0${sep}`;
+    assert.equal(statusBarAfterTab, `:e ${expectedPath}|`, '123');
+
+    // Tab to cycle the completion of tempDir
+    await modeHandler.handleKeyEvent('<tab>');
+    statusBarAfterTab = StatusBar.Get().trim();
+    expectedPath = `${tmpDir}${sep}inner1${sep}`;
+    assert.equal(statusBarAfterTab, `:e ${expectedPath}|`, '123');
+
+    // <right> and <tab> to select and complete the content in
+    // the inner1 directory
+    // Since there is only one directory in inner1 which is inner10,
+    // The completion is complete.
+    await modeHandler.handleMultipleKeyEvents(['<right>', '<tab>']);
+    statusBarAfterTab = StatusBar.Get().trim();
+    expectedPath = `${tmpDir}${sep}inner1${sep}inner10${sep}`;
+    assert.equal(statusBarAfterTab, `:e ${expectedPath}|`, '123');
+
+    // A tab would try to complete the content in the inner10.
+    // Since the pervious completion is complete, no <right> is needed to select
+    await modeHandler.handleKeyEvent('<tab>');
+    statusBarAfterTab = StatusBar.Get().trim();
+    expectedPath = `${tmpDir}${sep}inner1${sep}inner10${sep}inner100${sep}`;
+    assert.equal(statusBarAfterTab, `:e ${expectedPath}|`, '123');
+
+    // Since there isn't any files or directories in inner100, no completion.
+    await modeHandler.handleKeyEvent('<tab>');
+    statusBarAfterTab = StatusBar.Get().trim();
+    expectedPath = `${tmpDir}${sep}inner1${sep}inner10${sep}inner100${sep}`;
+    assert.equal(statusBarAfterTab, `:e ${expectedPath}|`, '123');
+
+    await modeHandler.handleKeyEvent('<Esc>');
+    await t.removeDir(inner100);
+    await t.removeDir(inner10);
+    await t.removeDir(inner1);
+    await t.removeDir(inner0);
+    await t.removeDir(tmpDir);
+  });
+
+  test('command line tab completion on the content on the left of the cursor', async () => {
+    await modeHandler.handleMultipleKeyEvents([':', 'e', 'd', 'i']);
+    await modeHandler.handleKeyEvent('<tab>');
+    let statusBarAfterTab = StatusBar.Get().trim();
+    assert.equal(statusBarAfterTab, ':edit|', 'Command Tab Completion Failed');
+
+    await modeHandler.handleMultipleKeyEvents(['<left>', '<left>']);
+    statusBarAfterTab = StatusBar.Get().trim();
+    assert.equal(statusBarAfterTab, ':ed|it', 'Failed to move the cursor to the left');
+
+    await modeHandler.handleKeyEvent('<tab>');
+    statusBarAfterTab = StatusBar.Get().trim();
+    assert.equal(statusBarAfterTab, ':edit|it', 'Failed to complete content left of the cursor');
+
+    await modeHandler.handleKeyEvent('<Esc>');
```

Comment:
Validate that we've exited tab completion?

Reply:
Not sure if we need to, and how.

Reply:
let's remove line 162 then ?

Reply:
If we don't have <ESC> at the end, it will messed up other tests.

Reply:
Tests shouldn't be dependent on each other. If this test fails causing other test failures, it will be difficult to deduce the cause.

Reply:
Sure. How about I call <Esc> always after a test case?